app/server/appsmith-git/src/main/resources/git.sh

```
@@ -50,6 +72,218 @@ git_download() {
+        echo "Error: '$directory_path' is not a git repository"
+        return 1
+    fi
+
+    # Verify the commit SHA exists in the target repository
+    if ! git -C "$directory_path" rev-parse --verify "$commit_sha^{commit}" > /dev/null 2>&1; then
+        echo "Error: Commit '$commit_sha' does not exist in repository at '$directory_path'"
+        return 1
+    fi
+
+    # Get the full SHA for better accuracy (in case a short SHA was provided)
+    local full_sha=$(git -C "$directory_path" rev-parse "$commit_sha")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)